libs/zephyr-agent/src/lib/build-context/ze-util-get-git-info.ts

```diff
@@ -123,3 +189,273 @@ function parseGitUrl(remoteOrigin: string, stdout: string) {
+        '⚠️  Git repository not found. Zephyr REQUIRES a git repository with remote origin.'
+      );
+      const promptResult = await promptForGitInfo(projectName);
+      org = promptResult.org;
+      project = promptResult.project;
+    } else {
+      // Non-interactive mode: use defaults and show warnings
+      logFn(
+        'error',
+        '⚠️  WARNING: Git repository not found. Zephyr REQUIRES git for proper deployment.'
+      );
+      logFn(
+        'error',
+        'Organization will be determined from your account - this is NOT recommended.'
+      );
+      ze_log.git('Organization will be determined from your account');
+      logFn(
+        'warn',
+        'To properly use Zephyr: git init && git remote add origin git@github.com:ORG/REPO.git'
+      );
+      logFn('info', 'Full setup guide: https://docs.zephyr-cloud.io');
+    }
+
+    const gitInfo: ZeGitInfo = {
+      git: {
+        name: name.trim(),
+        email: email.trim(),
+        branch: 'main',
+        commit: 'no-git-commit',
+        tags: [],
+      },
+      app: {
+        org,
+        project,
+      },
+    };
+
+    ze_log.git('Using global git config (no local repository)', gitInfo);
+
+    return gitInfo;
+  } catch (error) {
+    throw new ZephyrError(ZeErrors.ERR_NO_GIT_INFO, {
+      message: 'Failed to load global git config',
+      cause: error,
+    });
+  }
+}
+
+/** Try to get git info using environment variables for AI tools/platforms */
+async function getEnvBasedGitInfo(): Promise<ZeGitInfo | null> {
+  try {
+    // Check if we have the required environment variables
+    const org = process.env['ZEPHYR_ORG'];
+    const project = process.env['ZEPHYR_PROJECT'];
+
+    if (!org || !project) {
+      return null;
+    }
+
+    // Check if we have a valid token (required for authentication)
+    const token = await getToken();
+    if (!token || !isTokenStillValid(token, 60)) {
+      ze_log.git('Environment variables found but no valid token for authentication');
+      return null;
+    }
+
+    ze_log.git('Using environment variable fallback for deployment');
+    logFn('info', '🔧 Using ZEPHYR_ORG and ZEPHYR_PROJECT environment variables');
+
+    // Try to extract user information from the JWT token
+    let tokenUserName: string | undefined;
+    let tokenUserEmail: string | undefined;
+    let tokenUserId: string | undefined;
+
+    try {
+      const decodedToken = jose.decodeJwt(token);
+      ze_log.git('Decoded JWT token claims:', Object.keys(decodedToken));
+
+      // Zephyr uses both standard and namespaced claims
+      // Priority: namespaced claims > standard claims > other common claims
+      tokenUserName = (decodedToken['https://api.zephyr-cloud.io/name'] ||
+        decodedToken['name'] ||
+        decodedToken['nickname'] ||
+        decodedToken['preferred_username'] ||
+        decodedToken['given_name']) as string | undefined;
+
+      tokenUserEmail = (decodedToken['https://api.zephyr-cloud.io/email'] ||
```

Comment:
```suggestion
      tokenUserEmail = (decodedToken[TOKEN_EMAIL_KEY] ||
```

please use constants for this, and why the key is a url?